### PR TITLE
Dynamic Forms

### PR DESCRIPTION
**Pull request by Maximilian118**
## Implementing A Dynamic Form:

1. Add your form object to state.

2. Configure in what circumstances you'd like to render your form with the function at the bottom of utility.js. Note, here we can do much more than render depending on url. We can check for a token etc etc. 

3. Map through your form objects and apply the Input Component to each iteration.

## Code Layout:
- All form objects used in state are outsourced to forms.js.
- All form logic is in utility.js.

## Form Characteristics:

An input field will become invalid if ALL of the following are true:
- User input doesn't meet the RegEx requirements. (RegEx is in utility.js)
- User has touched/entered data into the input field. (Stops invalidation on page load)
- User has moved on to another input. (Stop invalidation while typing)

Submit button is disabled until all input fields are valid.

When an input becomes invalid it pushes an invalidMessage to state with more detail about why the input is invalid. 

## Available Input Types:

- Input
- Email
- Password
- TextArea
- File

## Notes:

I haven't included a select field as we don't need one yet nor do I see a future need for one.

### CODE REVIEW

File: src/components/UI/Button/Button.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
```
Comment (samuelbeard) [inline src/components/UI/Button/Button.js]: reactstrap already has a Button component

File: src/components/UI/Form/Form.module.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.Form {
```
Comment (samuelbeard) [inline src/components/UI/Form/Form.module.scss]: Use bootstraps form styling to keep everything uniform.

File: src/components/UI/Input/Input.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
+import React from "react"
+import * as classes from "./Input.module.scss"
+import PropTypes from 'prop-types'
+import { withStyles } from "@material-ui/core/styles"
```
Comment (samuelbeard) [inline src/components/UI/Input/Input.js]: Use bootstraps styling.

File: package.json [hunk 4]
```diff
@@ -3,6 +3,8 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@material-ui/core": "^4.3.2",
```
Comment (samuelbeard) [inline package.json]: We already have bootstrap. No need to add more styling.

File: src/components/UI/Input/Input.module.scss [hunk 5]
```diff
@@ -0,0 +1,71 @@
+.Input {
```
Comment (samuelbeard) [inline src/components/UI/Input/Input.module.scss]: Use bootstraps styling

File: src/components/UI/Button/Button.js (hunk 1 above)
Comment (Maximilian118) [reply src/components/UI/Button/Button.js]: Done

File: src/components/UI/Form/Form.module.scss (hunk 2 above)
Comment (Maximilian118) [reply src/components/UI/Form/Form.module.scss]: Dibe

File: src/components/UI/Input/Input.js (hunk 3 above)
Comment (Maximilian118) [reply src/components/UI/Input/Input.js]: Done

File: package.json (hunk 4 above)
Comment (Maximilian118) [reply package.json]: Done

File: src/components/UI/Input/Input.module.scss (hunk 5 above)
Comment (Maximilian118) [reply src/components/UI/Input/Input.module.scss]: Done

**Pull request merged**